frontend/src/store/modules/auth.js

```diff
@@ -1,42 +1,97 @@
 import axios from 'axios';
 
+import { interceptorRefresh } from '../../main.js';
+
 export default {
   namespaced: true,
   state: {
-    token: localStorage.getItem('token') || ''
+    token: localStorage.getItem('token') || '',
+    error: null
   },
   getters: {
-    isLoggedIn: state => !!state.token
+    isLoggedIn: state => !!state.token,
+    getError(state) {
+      return state.error;
+    }
   },
   mutations: {
-    auth_success(state, token) {
-      state.token = token;
+    raiseError(state, error_msg) {
+      state.error = error_msg;
     }
   },
   actions: {
     login({ commit }, loginData) {
-      return new Promise((resolve, reject) => {
-        axios
-          .post('auth/login/', {
-            username: loginData.username,
-            password: loginData.password
-          })
-          .then(
-            res => {
-              const token = res.data.access_token;
+      delete axios.defaults.headers.common['Authorization'];
+
+      return axios
+        .post('auth/login/', {
+          username: loginData.username,
+          password: loginData.password
+        })
+        .then(res => {
+          const token = res.data.access_token;
+          const refresh_token = res.data.refresh_token;
+
+          localStorage.setItem('token', token);
+          localStorage.setItem('refresh_token', refresh_token);
+          axios.defaults.headers.common['Authorization'] = 'Bearer ' + token;
+          return res.status;
+        })
+        .catch(error => {
+          localStorage.removeItem('token');
+          commit('raiseError', error);
+        });
+    },
+    logout({ dispatch }) {
+      axios.interceptors.response.eject(interceptorRefresh);
+      return axios
+        .delete('auth/logout/')
+        .then(() => {
+          localStorage.removeItem('token');
+        })
+        .catch(() => {
+          localStorage.removeItem('token');
+        })
+        .finally(() => {
+          delete axios.defaults.headers.common['Authorization'];
+          dispatch('revoke');
+        });
+    },
+    revoke() {
+      const refresh_token = localStorage.getItem('refresh_token');
```

Comment:
Access token is added as a default header in main.js, because we have to provide it in every request to endpoint required authentication. Refresh token is required only to refresh access token and to revoke itself. Since header for both access and refresh token is these same:
"Authorization": "Bearer {access_token}"
"Authorization": "Bearer {refresh_token}"
setting it with refresh_token doesn't make much sense. What is more important, we should consider different strategy to store refresh_token (I mentioned it on few hacknights). Refresh token lives long, so it is important to store it and use it safely. Here is one of possible solutions https://flask-jwt-extended.readthedocs.io/en/stable/tokens_in_cookies/